run/awake_start

```diff
@@ -2,13 +2,82 @@
 
 if [[ ( -z "${TESLAFI_API_TOKEN:+x}" ) && ( -z "${TESSIE_API_TOKEN:+x}" ) && ( -z "${TESLA_BLE_VIN:+x}" ) ]]
 then
-  log "Not keeping car awake using Sentry Mode."
+  log "Not configured to keep car awake."
 else
+  function nudge {
+    local retry=1
+
+    while true
+    do
+      if [[ -n "${TESSIE_API_TOKEN:+x}" ]]
+      # Use Tessie API to nudge the car
+      then
+        http_response=$(curl -s -w "%{http_code}" \
+          -H "Authorization: Bearer $TESSIE_API_TOKEN" \
+          -H "Accept: application/json" \
+          -H "User-Agent: github.com/marcone/teslausb" \
+          "https://api.tessie.com/$TESSIE_VIN/command/close_charge_port")
+
+        log "Tessie: Nudging the car."
+        # Extract HTTP status code
+        http_status="${http_response: -3}"
+        
+        # Check if HTTP status code is not 200 (OK)
+        if [ "$http_status" != "200" ]
+        then
+          # Log an error
+          log "Tessie: Failed to send command. Check your TESSIE_API_TOKEN and TESSIE_VIN are set correctly. If they are, your car may be offline or asleep."
+          if [ "$retry" -lt "3" ]
+          then
+            log "Tessie: Retrying after 30s in case the connection was poor."
+            retry=$((retry+1))
+            sleep 30
+            continue
+          else
+            log "Tessie: Connection to the internet likely broken."
+            break
+          fi
+        fi
+      fi
+
+      if [[ -n "${TESLA_BLE_VIN:+x}" ]]
+      #Use Tesla BLE API to nudge the car
```

Comment:
Nitpick: there are a few places like this where there's no space after the `#`, which is less readable and doesn't match other comments.